### PR TITLE
Feature get top users

### PR DESCRIPTION
**Pull request by jwahjudi**

### CODE REVIEW

File: src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java [hunk 1]
```diff
@@ -124,6 +128,50 @@ public User addUser(PrimaryKey id) {
...
+		
+		LinkedHashMap<Integer, Integer> uniqueId = new LinkedHashMap<>();
+		for(User user: users) {
+			if(!uniqueId.containsKey(user.getId().getUserId())) {
+				uniqueId.put(user.getId().getUserId(), 1);
+			} else {
+				uniqueId.put(user.getId().getUserId(), uniqueId.get(user.getId().getUserId()) + 1);
+			}
+		}
+		System.out.println("hello " + users);
```
Comment (mosopeogundipe) [inline src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java]: Remove print statement used for debugging

File: src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java [hunk 2]
```diff
@@ -124,6 +128,50 @@ public User addUser(PrimaryKey id) {
...
+		List<Entry<Integer,Integer>> result = new ArrayList<>(uniqueId.entrySet());
+		result.sort(Entry.comparingByValue());
+		Collections.reverse(result);
+		if(page*nums-1<=result.size()) {
```
Comment (mosopeogundipe) [inline src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java]: This wouldn't work if page = 0. 
In that case, the result populated [here ](https://github.com/usfcs-perf-eng-s20/heroku-project-favs/pull/8/files#diff-0fae860ae0c1e35be827d26bc884c318R156) will most likely be an empty list

Whereas Page =0 is a valid value. Maybe have a special logic that handles page=0

File: src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java (hunk 1 above)
Comment (mosopeogundipe) [inline src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java]: Remove print statement used for debugging

**Review by mosopeogundipe — CHANGES_REQUESTED** [review]

**Comment by kedarkhetia** [comment]
Can we also squash all the commits to 1?

Something like this. 
https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

I see a lot of dead code in lot of these commits. You might want to squash them all and then make a pull request with relevant commit message.

**Review by mosopeogundipe — APPROVED** [review]

**Review by kedarkhetia — CHANGES_REQUESTED** [review]
Can we also squash all the commits to 1?

Something like this.
https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

I see a lot of dead code in lot of these commits. You might want to squash them all and then make a pull request with relevant commit message.

Also remove .DS_Store files (you can do that by adding them into .gitignore). Also, remove the target class file. It seems that your base branch is behind master because of which you are facing these conflicts with master.

**Review by kedarkhetia — CHANGES_REQUESTED** [review]
I see a lot of changes that are already present in the master (from other authors). We should avoid adding duplicate code as much as possible. Especially since these changes are part of your CR, for the features you were never assigned, I would not be comfortable approving it. 

My suggestion would be to just deprecate this branch. Create a new branch from the current master branch as base and redo your portion of the changes in it. After you verify create a new pull request and we can take a look.

**Comment by mosopeogundipe** [comment]
> 
> 
> I see a lot of changes that are already present in the master (from other authors). We should avoid adding duplicate code as much as possible. Especially since these changes are part of your CR, for the features you were never assigned, I would not be comfortable approving it.
> 
> My suggestion would be to just deprecate this branch. Create a new branch from the current master branch as base and redo your portion of the changes in it. After you verify create a new pull request and we can take a look.

Yes, I agree with this. @jwahjudi let's try to:
1) Create a new branch based on master 
2) Copy/paste your changes into the new branch
3) Make a pull request based for the changes you added in (2) above

That will both solve ensure you can merge seamlessly and also wouldn't ascribe code changes done by others to you.

**Comment by jwahjudi** [comment]
I squash the debugging codes, but for some reason

**Pull request closed without merging**